internal/caddy/ingress/redirect.go

```diff
@@ -0,0 +1,90 @@
+package ingress
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/caddyserver/caddy/v2/caddyconfig"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	"github.com/caddyserver/ingress/pkg/converter"
+)
+
+type RedirectPlugin struct{}
+
+func (p RedirectPlugin) IngressPlugin() converter.PluginInfo {
+	return converter.PluginInfo{
+		Name:     "ingress.redirect",
+		Priority: 10,
+		New:      func() converter.Plugin { return new(RedirectPlugin) },
+	}
+}
+
+// IngressHandler Converts redirect annotations to static_response handler
+func (p RedirectPlugin) IngressHandler(input converter.IngressMiddlewareInput) (*caddyhttp.Route, error) {
+	ing := input.Ingress
+
+	permanentRedir := getAnnotation(ing, permanentRedirectAnnotation)
+	temporaryRedir := getAnnotation(ing, temporaryRedirectAnnotation)
+
+	var code string = "301"
+	var redirectTo string = ""
+
+	// Don't allow both redirect annotations to be set
+	if permanentRedir != "" && temporaryRedir != "" {
+		return nil, fmt.Errorf("cannot use permanent-redirect annotation with temporal-redirect")
+	}
+
+	if permanentRedir != "" {
+		// Logic taken from Caddy's parseRedir builtin function
+		redirectCode := getAnnotation(ing, permanentRedirectCodeAnnotation)
+		if redirectCode != "" {
+			switch redirectCode {
+			case "permanent":
+				code = "301"
+			case "temporary":
+				code = "302"
```

Comment:
Do you think we still need it? As now we have specific annotation for permanent and temporary redirect, I suppose we can get rid of them.

Reply:
I've removed them now.